packages/http/fetch/src/middlewares/options/headersInspectionOptions.ts

```diff
@@ -27,18 +48,35 @@ export class HeadersInspectionOptions implements RequestOption {
 	public getResponseHeaders() {
 		return this.responseHeaders;
 	}
+
+	/**
+	 * @public
+	 * @member
+	 * @default false
+	 * whether to inspect request headers
+	 */
+	public inspectRequestHeaders: boolean;
+
+	/**
+	 * @public
+	 * @member
+	 * @default false
+	 * whether to inspect response headers
+	 */
+	public inspectResponseHeaders: boolean;
+
 	/**
 	 * @public
 	 * @constructor
 	 * To create an instance of HeadersInspectionOptions
-	 * @param {boolean} [inspectRequestHeaders = false] - Whether to inspect request headers
-	 * @param {boolean} [inspectResponseHeaders = false] - Whether to inspect response headers
+	 * @param {HeadersInspectionOptionsParams} [options = {}] - The headers inspection options value
 	 * @returns An instance of HeadersInspectionOptions
 	 */
-	public constructor(
-		public inspectRequestHeaders = false,
-		public inspectResponseHeaders = false,
-	) {}
+	public constructor(options: HeadersInspectionOptionsParams = {} as HeadersInspectionOptionsParams) {
```

Comment:
as far as I understand this will request people to either pass nothing, or the whole set of properties.
Could we type the argument with `Partial<HeadersInspectionOptionsParams>` instead?